File: Data Structures & Algorithms/two-integer-sum/submission-4.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp ;
        vector<int> sol;
        for(int i= 0;i<nums.size();i++)
        {
            int diff=target-nums[i];
            if(mp.find(diff)!=mp.end())
            {
                // sol.push_back(i);
                sol.push_back(mp[diff]);
                sol.push_back(i);
                return sol;
            }
            else
            {
                mp[nums[i]]=i;
            }
        }
        return sol;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-4.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[1]*len(nums)
        p=1
        for i in range(len(nums)):
            p=p*nums[i]
            res[i]=p
        p=1
        for i in range(len(res)-1, 0 ,-1):
           res[i]=res[i-1] * p
           p=p*nums[i]
        res[0]=p
        return res